ibm/resource_ibm_pi_image_test.go

```diff
@@ -98,3 +102,37 @@ func testAccCheckIBMPIImageConfig(name string) string {
 	  }
 	`, name, pi_cloud_instance_id)
 }
+
+func TestAccIBMPIImageCOSPublicImport(t *testing.T) {
+	imageRes := "ibm_pi_image.cos_image"
+	name := fmt.Sprintf("tf-pi-image-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckIBMPIImageDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMPIImageCOSPublicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckIBMPIImageExists(imageRes),
+					resource.TestCheckResourceAttr(imageRes, "pi_image_name", name),
+					resource.TestCheckResourceAttrSet(imageRes, "image_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMPIImageCOSPublicConfig(name string) string {
+	return fmt.Sprintf(`
+	resource "ibm_pi_image" "cos_image" {
+		pi_image_name       = "%s"
+		pi_cloud_instance_id = "%s"
+		pi_image_bucket_name = "ocp4-images-public-bucket"
+		pi_image_bucket_access = "public"
+		pi_image_bucket_region = "us-south"
+		pi_image_bucket_file_name = "rhcos-48-07222021.ova.gz"
```

Comment:
instead of hardcoding the values pi_image_bucket_name and filename...can be add them to provider_test.go something similar to "pi_cloud_instance_id" so that other can set the values

Reply:
done